### DISCUSSION TITLE
Emit function after "close" modal

### DISCUSSION THREAD

**Discussion by LaMance22** [wireui/wireui#419] [Q&A] [answered]
Hi, is it possible to emit a function from "livewire" after modal close? I can add "wire:click" to button, but closing modal is possible by clicking outside window. I want to "reset" all fields after closing modal. Thanks for help.

**Comment by ph7jack** [comment, marked as answer]
@sonets2017 It's possible, look at the documentation 
https://wireui-docs.test/docs/modal#modal-events 
Or do it using the livewire hooks
https://laravel-livewire.com/docs/2.x/lifecycle-hooks

**Comment by LaMance22** [comment]
Thanks! Its works. I emit function after close modal and reset all fields.
`x-on:close="Livewire.emit('resetAll')"`
